src/python/pants/util/dirutil.py

```diff
@@ -297,16 +297,25 @@ def register_rmtree(directory, cleaner=_mkdtemp_atexit_cleaner):
   return directory
 
 
-def safe_rmtree(directory):
-  """Delete a directory if it's present. If it's not present, no-op.
+def append_os_sep(directory):
+  return os.path.join(directory, directory + os.sep)
+
 
-  Note that if the directory argument is a symlink, only the symlink will
-  be deleted.
+def safe_rmtree(directory, recursive=False):
+  """Delete a directory if it's present. If it's not present, no-op.
+  :param directory: path to a directory or a symlink
+  :param recursive: when set to False (default) and directory passed is a symlink, only the symlink
+                    will be deleted. When set to True, follow a symlink and delete everything
+                    inside.
 
   :API: public
   """
   if os.path.islink(directory):
-    safe_delete(directory)
+    if recursive:
+      directory = append_os_sep(directory)
+      shutil.rmtree(directory, ignore_errors=True)
```

Comment:
If you take the above recommendation to inline `append_os_sep`, I would consolidate these two lines to be simply
```python
shutil.rmtree(directory + os.sep, ignore_errors=True)
```